hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionEventManager.java

```diff
@@ -79,25 +83,25 @@ public void sendMigrationEvent(MigrationState state, MigrationInfo migrationInfo
     }
 
     public void sendMigrationProcessStartedEvent(MigrationState state) {
-        ReplicaMigrationEvent event = new ReplicaMigrationEventImpl(state, MIGRATION_STARTED, 0, null, null, false, 0L);
+        ReplicaMigrationEvent event =
+                new ReplicaMigrationEventImpl(state, MIGRATION_STARTED_PARTITION_ID, 0, null, null, false, 0L);
         sendMigrationEvent(event);
     }
 
     public void sendMigrationProcessCompletedEvent(MigrationState state) {
-        ReplicaMigrationEvent event = new ReplicaMigrationEventImpl(state, MIGRATION_FINISHED, 0, null, null, false, 0L);
+        ReplicaMigrationEvent event =
+                new ReplicaMigrationEventImpl(state, MIGRATION_FINISHED_PARTITION_ID, 0, null, null, false, 0L);
         sendMigrationEvent(event);
     }
 
     private void sendMigrationEvent(ReplicaMigrationEvent event) {
         EventService eventService = nodeEngine.getEventService();
         // All migration events are sent in order.
-        eventService.publishEvent(SERVICE_NAME, MIGRATION_EVENT_TOPIC, event, MIGRATION_EVENT_TOPIC.hashCode());
+        eventService.publishEvent(SERVICE_NAME, MIGRATION_EVENT_TOPIC, event, MIGRATION_EVENT_TOPIC_ORDER_KEY);
```

Comment:
Another stupid question for @sancar. Why don't we use `event.getPartitionId()` instead of `MIGRATION_EVENT_TOPIC_ORDER_KEY` here?

A side note. I don't really like this implicit logic based on hash code for event order key. We may easily run into collisions between different event types without even noticing it.

Reply:
I dare to answer this question :D

If you take a look at lines 86 and 92
You can see that when the migration process is started or finished we send accordingly `partitionId` = -1 if started or -2 if finished. But on typical `ReplicaMigrationEvent` we send `partitionId` from `MigrationInfo` (line 75) .

To ensure that these events are sent in the correct order, they should have the same `orderKey`

Reply:
> If you take a look at lines 86 and 92
> You can see that when the migration process is started or finished we send accordingly `partitionId` = -1 if started or -2 if finished. But on typical `ReplicaMigrationEvent` we send `partitionId` from `MigrationInfo` (line 75) .

Makes sense. So, the reason is those global events.

As for the collisions, I've checked the local branch which boils down to `StripedExecutor#getWorker` and it seems that collisions are tolerated by the design.